test/integration/federation/crud_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package federation
+
+import (
+	"testing"
+
+	"github.com/pborman/uuid"
+
+	"k8s.io/kubernetes/test/integration/federation/framework"
+)
+
+// TestFederationCRUD validates create/read/update/delete operations for federated resource types.
+func TestFederationCRUD(t *testing.T) {
+	fedFixture := framework.FederationFixture{ClusterCount: 2}
+	fedFixture.SetUp(t)
+	defer fedFixture.TearDown(t)
+
+	controllerFixtures := []framework.ControllerFixture{
+		&framework.SecretFixture{},
+	}
+	for _, fixture := range controllerFixtures {
+		kind := fixture.Kind()
+		t.Run(kind, func(t *testing.T) {
+			framework.SetUpControllerFixture(t, fedFixture.APIFixture, fixture)
+			defer fixture.TearDown(t)
+
+			crud := framework.NewCRUDHelper(t, fixture.Adapter(), fedFixture.ClusterClients)
+			obj := fixture.NewObject(uuid.New())
+			crud.CheckLifecycle(obj)
+		})
+	}
+
+	// Validate deletion handling where orphanDependents is true or nil for a single resource type since the
+	// underlying logic is common across all types.
+	orphanedDependents := true
+	testCases := map[string]*bool{
+		"Resources should not be deleted from underlying clusters when OrphanDependents is true": &orphanedDependents,
+		"Resources should not be deleted from underlying clusters when OrphanDependents is nil":  nil,
+	}
+	for testName, orphanDependents := range testCases {
+		t.Run(testName, func(t *testing.T) {
+			fixture := &framework.SecretFixture{}
```

Comment:
Was this the unresolved comment?  I did actually answer in reviewable:

```
This PR was originally targeting daemonset and configmap as well as secrets, and I'll be posting PRs for those types as soon as this PR merges.
```

What's more likely is that post-controller refactor a collection of type adapters is used to configure controllers and testing rather than having to specify the type adapters explicitly.  This is under heavy enough development that I don't see a need to document future intent.